Assets/MixedRealityToolkit/_Core/Utilities/Lines/DataProviders/RectangleLineDataProvider.cs

```diff
@@ -0,0 +1,157 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Internal.Utilities.Lines.DataProviders
+{
+    public class RectangleLineDataProvider : BaseMixedRealityLineDataProvider
+    {
+        [SerializeField]
+        private Vector3[] points = new Vector3[8];
+
+        [SerializeField]
+        private float width = 1f;
```

Comment:
ranges?

Reply:
wasn't in the original, but yeah probably.

Reply:
After looking into this, a range wouldn't be helpful. We don't know how large users may make this width, but we do make sure that if it's under 0, we set it to the current width /height, so there's that.

I suppose I could make another custom attribute to make sure it stays above 0.